api/src/lib.rs

```diff
@@ -84,7 +84,15 @@ pub async fn submit_request(input: HttpRequest) -> Result<Value, Box<dyn Error>>
         HttpMethod::HEAD => todo!(),
     };
 
+    let res_type = res.headers().get("content-type").unwrap().to_str().unwrap();
+
+    if !res_type.starts_with("application/json") {
+        // I couldn't figure out how to safely throw an error so I'm just returning this for now
+        println!("expected application/json, got {}", res_type);
+        return Ok(json!({"msg": "not JSON!"}));
```

Comment:
this is 100% me being stupid, spent like an hour trying to figure out how to safely throw an error here and gave up, decided to just return a normal JSON value for now so it doesn't break the UI

Reply:
I think I ran into an issue with trying to throw an error too, relating to `Box<dyn Error>` not implementing Sync or Send or something like that I think? Obv have some refreshing to do there hehe